server/oauth_migration.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/mattermost/mattermost-plugin-api/cluster"
+	"github.com/pkg/errors"
+)
+
+const (
+	oauthMigrationStoreKey = "oauth_migration"
+	oauthMigrationMutexKey = "oauth_migration_mutex"
+
+	oauthMigrationStatusInProgress = "IN_PROGRESS"
+	oauthMigrationStatusComplete   = "COMPLETE"
+	oauthMigrationStatusError      = "ERROR"
+)
+
+func (p *Plugin) checkAndPerformOAuthTokenMigration() error {
+	mutexAPI := cluster.MutexPluginAPI(p.API)
+	mutex, err := cluster.NewMutex(mutexAPI, oauthMigrationMutexKey)
+	if err != nil {
+		return err
+	}
+
+	mutex.Lock()
+	defer mutex.Unlock()
+
+	var status string
+	err = p.client.KV.Get(oauthMigrationStoreKey, &status)
+	if err != nil {
+		return err
+	}
+
+	// Migration is in progress or already completed
+	if status != "" {
+		return nil
+	}
+
+	_, _ = p.client.KV.Set(oauthMigrationStoreKey, oauthMigrationStatusInProgress)
```

Comment:
nit: should we check the err here?

Reply:
@spirosoik Logging those errors now. Please take another look if you'd like